pkg/formula/runner/pre_run_builder.go

```
@@ -47,7 +47,8 @@ func NewPreRunBuilder(
 func (b PreRunBuilderManager) Build(relativePath string) {
```

I think you could return an error in this method, it will make your tests easier.